pyxem/signals/polar_diffraction2d.py

```diff
@@ -139,30 +141,32 @@ def get_pearson_correlation(self, mask=None, krange=None, inplace=False, **kwarg
             The pearson rotational correlation when inplace is False, otherwise
             return None
         """
-        if krange is None:
-            correlation = self.map(_pearson_correlation, mask=mask, inplace=inplace, **kwargs)
-        else:
+        # placeholder to handle inplace playing well with cropping and mapping
+        s_ = self
+        if krange is not None:
             if inplace:
-                raise ValueError("`krange` is not compatible with `inplace=True`")
-            k_range_slice = self.isig[:, krange[0]:krange[1]]
-            if mask is not None:
-                mask_signal = Signal2D(mask)
-                mask_signal.axes_manager.signal_axes[1].scale = self.axes_manager[-1].scale
-                mask_signal.axes_manager.signal_axes[1].offset = self.axes_manager[-1].offset
-                mask_slice = mask_signal.isig[:, krange[0]:krange[1]]
-                correlation = k_range_slice.map(_pearson_correlation,
-                                                mask=mask_slice,
-                                                inplace=inplace,
-                                                **kwargs)
+                s_.crop(-1, start=krange[0], end=krange[1])
             else:
-                correlation = k_range_slice.map(_pearson_correlation,
-                                                inplace=inplace,
-                                                **kwargs)
+                s_ = self.isig[:, krange[0]:krange[1]]
```

Comment:
Why is crop used for inplace and isig used for not inplace?

Reply:
`crop` works inplace, while `isig` return a separate signal, and they are used in the corresponding case - very much as you said in your question! ;)